dbt_style_guide.md

```diff
@@ -1,237 +1,522 @@
+          nations.nation,
+          nations.nation_comment
+      from regions
+      left join nations
+          on regions.region_id = nations.region_id
+  ),
+  
+  final as (
+      select
+          suppliers.supplier_id,
+          suppliers.location_id,
+          locations.region_id,
+          locations.nation_id,
+          suppliers.supplier_name,
+          suppliers.supplier_address,
+          suppliers.phone_number,
+          locations.region,
+          locations.region_comment,
+          locations.nation,
+          locations.nation_comment,
+          suppliers.account_balance
+      from suppliers
+      inner join locations
+          on suppliers.location_id = locations.location_sk
+  )
+  
+  -- Simple select statement
+  select * from final
+  ```
 
 ## SQL style guide
+- **DO NOT OPTIMIZE FOR FEWER LINES OF CODE.**  
+
+  New lines are cheap, brain time is expensive; new lines should be used within  
+  reason to produce code that is easily read.
 
 - Use trailing commas
-- Indents should be four spaces (except for predicates, which should line up with the `where` keyword)
-- Lines of SQL should be no longer than 80 characters
-- Field names and function names should all be lowercase
-- The `as` keyword should be used when aliasing a field or table
-- Fields should be stated before aggregates / window functions
-- Aggregations should be executed as early as possible before joining to another table.
-- Ordering and grouping by a number (eg. group by 1, 2) is preferred over listing the column names (see [this rant](https://blog.getdbt.com/write-better-sql-a-defense-of-group-by-1/) for why). Note that if you are grouping by more than a few columns, it may be worth revisiting your model design.
-- Prefer `union all` to `union` [*](http://docs.aws.amazon.com/redshift/latest/dg/c_example_unionall_query.html)
-- Avoid table aliases in join conditions (especially initialisms) – it's harder to understand what the table called "c" is compared to "customers".
-- If joining two or more tables, _always_ prefix your column names with the table alias. If only selecting from one table, prefixes are not needed.
-- Be explicit about your join (i.e. write `inner join` instead of `join`). `left joins` are normally the most useful, `right joins` often indicate that you should change which table you select `from` and which one you `join` to.
 
-- *DO NOT OPTIMIZE FOR A SMALLER NUMBER OF LINES OF CODE. NEWLINES ARE CHEAP, BRAIN TIME IS EXPENSIVE*
+- Indents should use four spaces. 
+
+- When dealing with long `when` or `where` clauses, predicates should be on a new
+  line and indented.  
+  Example:
+  ```sql
+  where 
+      user_id is not null
+      and status = 'pending'
+      and location = 'hq'
+  ```
+
+- Lines of SQL should be no longer than 80 characters and new lines should be used to ensure this.  
+  Example:
+  ```sql
+  sum(
+      case when order_status = 'complete' then order_total end
+  ) as monthly_total,

```

Comment:
what abt this?

I like having the `case when` inputs as individual lines for readability purposes.
we do the same for window functions so this feeeeels right?

```suggestion
  sum(
      case
          when order_status = 'complete'
              then order_total 
      end
  ) as monthly_total,

```

Reply:
I think this is a fine ask as it follows our other best practices! I think an hidden rule here is that if it doesn't exceed 80 characters, it's fair game so you're free to be concise, but I think for the purposes of example this should be strict.